commands/up.go

```diff
@@ -64,20 +77,133 @@ func preRunUp(cmd *cobra.Command, args []string) error {
 }
 
 func upHandler(cmd *cobra.Command, args []string) error {
-	if err := runBuild(cmd, args); err != nil {
+	handler := upRunner(cmd, args)
+	if err := handler(); err != nil {
 		return err
 	}
-	fmt.Println()
-	if !skipPush {
-		if err := runPush(cmd, args); err != nil {
+
+	if watch {
+		logger := log.New(os.Stderr, "", log.LstdFlags)
+		logger.SetFlags(log.Ldate | log.Ltime | log.LUTC)
+
+		watcher, err := fsnotify.NewWatcher()
+		if err != nil {
 			return err
 		}
-		fmt.Println()
-	}
-	if !skipDeploy {
-		if err := runDeploy(cmd, args); err != nil {
+		defer watcher.Close()
+
+		patterns, err := ignorePatterns()
+		if err != nil {
+			return err
+		}
+
+		if err = filepath.Walk("./", func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+
+			for _, pattern := range patterns {
+				match, err := filepath.Match(pattern, path)
```

Comment:
If we will use it.

For example:

The **`.gitignore`** file contains **`build`**.

Suppose there is a function named **`gobuilder`** or something containing **`build`**, then the watcher will skip watching the files in the **`gobuilder`** folder too. This is what we do not want to happen. Therefore I think we should skip the folders exactly mentioned in the **`.gitignore`** because we can afford to watch a not required file but cannot skip watching a required file.

Please give your thoughts, there might be something I am missing here or unable to understand your suggestion.

Reply:
hmm, i wonder how git handles this, maybe they do something like "if the pattern contains no stars, test if the path base == the pattern", something like this https://pkg.go.dev/path/filepath#Base

```go
match = match || filepath.Base(path) == pattern
```

Reply:
@LucasRoesler Could you please give an example of a pattern that we want to skip but not skipped by current filepath.Match(pattern, path) ?

Reply:
@LucasRoesler According to you "if the pattern contains no stars, test if the `path base == the pattern`"

Suppose pattern with no `*` is `example.txt` and path is `build/example.txt`. Path base for this case will be `example.txt`.

* If we use `filepath.Base(path) == pattern`, then path `build/example.txt` will match the pattern `example.txt`, which is the wrong behavior. 

* But if this case is handled by the `filepath.Match(pattern, path)`, in this case, the pattern will only be matched with the exact path given with the current implementation, which will be the correct behavior.

Could you please give an example of a pattern that we want to skip but is not taken into account by the current `filepath.Match(pattern, path)` implementation?

Reply:
the reason i am thinking of this example 
```
.venv
__pycache__
*.pyc

logs/

.env
keyfile.json
.task
```

which is a mix of patterns and non-patterns. The `__pycache__` is something that could easily appear inside the template path and git would ignore it and i think we would want the watch to ignore it as well.

Reply:
Have you looked at go-git - https://github.com/go-git/go-git/blob/dc2b346ed149080199d578f0190f3ac1156480c2/plumbing/format/gitignore/dir.go

Reply:
Likewise, the git docs describe the format in detail - https://git-scm.com/docs/gitignore#_pattern_format

Reply:
Thanks for the comment John. I've sent you a Twitter DM if you still have an account and are interested in keeping in touch.

Reply:
@LucasRoesler @johnmccabe Now I am using **"github.com/go-git/go-git/v5/plumbing/format/gitignore"** package for matching the patterns from `.gitignore`. Could you please take a look?